### DISCUSSION TITLE
File Not Found

### DISCUSSION THREAD

**Discussion by 250king** [Mrs4s/go-cqhttp#2375] [Q&A] [answered]
这几天迁移bot到Linux，结果原本正常的图片发送抽风了
![Screenshot_2023-08-08-17-27-25-54](https://github.com/Mrs4s/go-cqhttp/assets/61256966/e4326ec2-f4cb-48c5-85f4-3d63ba035c98)
日志显示File Not Found，但文件存在呀（

**Comment by 1umine** [comment]
确定真的存在吗（），`ls /root/resource/out` 看看

**Reply by 250king** [reply]
![Screenshot_2023-08-16-00-15-01-27](https://github.com/Mrs4s/go-cqhttp/assets/61256966/3fccc590-dc8a-48f8-8761-762d48f0b631)

**Reply by 250king** [reply]
这是今天的，也是同样问题

**Reply by 1umine** [reply]
感觉可能是权限问题，可能是gocq权限不足，看看发送失败时的gocq日志（？

**Reply by 250king** [reply]
权限给了也是有问题
![Screenshot_2023-08-16-22-43-01-82](https://github.com/Mrs4s/go-cqhttp/assets/61256966/9a71c43e-d62a-477d-8e26-e1b7e0316d42)
不知道是不是本身路径格式有问题，因为好像少了个/，即正确写法为file:////root/resource/out/387665635_2789260300.png

**Reply by 1umine** [reply]
> 不知道是不是本身路径格式有问题，因为好像少了个/，即正确写法为file:////root/resource/out/387665635_2789260300.png

感觉不应该，相信pathlib.Path（

路径正确文件存在，在同一台机器上理论上是没问题的

gocq是否是在容器内部署的（
看你的hostname好像是进了容器内？文件在容器里，但是gocq在外面或者其他容器内？

**Reply by 250king** [reply]
已经挂在同一个volume了（
![Screenshot_2023-08-20-21-57-47-53](https://github.com/Mrs4s/go-cqhttp/assets/61256966/90787215-02e6-43dc-9239-41aae64d1d4b)

**Reply by 250king** [reply, marked as answer]
刚刚检查了一下，发现go-cqhttp以别的身份运行，即ID为99的用户
也就是说因为权限问题无法访问
后来试着修改docker-entrypoint.sh，让go-cqhttp以root身份运行，现在可以了
